openapi3/loader_recursive_ref_test.go

```diff
@@ -47,3 +48,9 @@ components:
 		// type
 		Value.Type)
 }
+
+func TestIssue570(t *testing.T) {
+	loader := NewLoader()
+	_, err := loader.LoadFromURI(&url.URL{RawPath: "https://rubrikinc.github.io/api-doc-internal-6.0/openapi.json"})
```

Comment:
I'd suggest you first expand this link into its JSON then prune parts of that document until the infinite loop issue disappears. Then you'll have a minimized regression test and should be able to actually solve this from here :)

Reply:
Yeah, you're absolutely right. I spent over 3 hours trying to do just that prior to submitting this commit (pruning to a minimum) but unfortunately I was unable to succeed, and could spend more time on this.. This is a gigantic file, and every time I thought it should happen, the problem didn't occur.. I left this in to at least show the problem. 

I know that as it is, this pr cannot be merged, hoping we could locate a better solution from the other pr you marked this as duplicate of. 

Thanks!!